envbuilder.go

```diff
@@ -276,407 +276,421 @@ func Run(ctx context.Context, opts options.Options) error {
 		}
 	}
 
-	if buildParams == nil {
-		// If there isn't a devcontainer.json file in the repository,
-		// we fallback to whatever the `DefaultImage` is.
-		var err error
-		buildParams, err = defaultBuildParams()
-		if err != nil {
-			return fmt.Errorf("no Dockerfile or devcontainer.json found: %w", err)
+	var (
+		username       string
+		skippedRebuild bool
+	)
+	if _, err := os.Stat(magicDir.Image()); errors.Is(err, fs.ErrNotExist) {
```

Comment:
nit: that's a huge condition :) but I'm afraid it matches the rest of the codebase.

Reply:
Yes, agreed. Some refactoring in the future would be on-point. 😄